src/components/ChattingScreen.js

```diff
@@ -1,31 +1,63 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 
 import Header from './Header';
 import MessageSender from './MessageSender';
+import ChattingMessage from './ChattingMessage';
 
-export default function ChattingScreen() {
-	const EUNKO = 'https://img.techpowerup.org/200908/eun.png';
-	const COOL = 'https://img.techpowerup.org/200908/NjRiY2JjOGU5YzQz.png';
-
-	const MSGLIST = [
-		{ user: true, content: '안녕하세요 12기 프론트엔드 개발자분들' },
-		{ user: true, content: '저희의 대화를 마음껏 조작해보세요 💌' },
-		{ user: true, content: '상단에 프로필을 눌러서 발신자 변경하면 됩니당~' },
-		{ user: false, content: '안녕 은아' },
-		{ user: false, content: '뭐해 ?' },
-		{ user: true, content: '시원아 넌 최고의 팀장이야' },
-		{ user: false, content: '나도 아니까  ' },
-		{ user: false, content: '그만 말해줘도 돼' },
-		{ user: true, content: 'ㅠㅠ' },
-	];
-
-	return (
-		<Wrapper>
-			<Header></Header>
-			<MessageSender></MessageSender>
-		</Wrapper>
-	);
-}
-
-const Wrapper = styled.div``;
+const ChattingScreen = () => {
+    const EUNKO = 'https://img.techpowerup.org/200908/eun.png';
+    const COOL = 'https://img.techpowerup.org/200908/NjRiY2JjOGU5YzQz.png';
+    const MESSAGELIST = [
+        { user: true, content: '안녕하세요 12기 프론트엔드 개발자분들' },
+        { user: true, content: '저희의 대화를 마음껏 조작해보세요 💌' },
+        { user: true, content: '상단에 프로필을 눌러서 발신자 변경하면 됩니당~' },
+        { user: false, content: '안녕 은아' },
+        { user: false, content: '뭐해 ?' },
+        { user: true, content: '시원아 넌 최고의 팀장이야' },
+        { user: false, content: '나도 아니까  ' },
+        { user: false, content: '그만 말해줘도 돼' },
+        { user: true, content: 'ㅠㅠ' },
+    ];
+    // false: cool  true: eunko
+    const userList = [COOL, EUNKO];
+    const [status, setStatus] = useState(true);
+    const [messageList, setMessageList] = useState(MESSAGELIST);
+
+    useEffect(() => {
+        window.scrollBy(0, document.body.scrollHeight);
+    }, [messageList]);
+
+    // switch User function
+    const toggleUser = () => {
+        setStatus(!status);
+    };
+
+    const onChangeMessageList = (newMessage) => {
+        setMessageList(messageList.concat(newMessage));
+        // 왜 push는 안되는지 모르겠습니다...
+    };
+
+    return (
+        <Screen>
+            <Header userList={userList} toggleUser={toggleUser} status={status}></Header>
```

Comment:
여기에서 userList를 넘겨주는 게 아니라 현재 status에 맞게 분기처리해서 그에 해당하는 값만 넘기면 Header.js 에서 state를 따로 선언해서 분기처리하는 함수를 만들 필요없이 바로 값을 집어넣을 수 있을 것 같아요
```suggestion
            <Header userName={status ? '고은' : '정쿨'} userImg={status ? EUNKO : COOL} toggleUser={toggleUser}></Header>
```

Reply:
state를 묶어서 넘겨주는것 보다 이렇게 하나씩 조건에 맞게 넘겨주는게 편할 때도 있군요!! 감사합니다 ㅠㅠ